lambdas/handlers/create_document_reference_handler.py

```diff
@@ -41,141 +34,48 @@ def lambda_handler(event, context):
-            for doc in body["content"][0]["attachment"]
-        ]
+    except CreateDocumentRefException as e:
+        logger.error(
+            e.message, {"Result": f"Upload Unsuccessful due to {str(e.message)}"}
+        )
 
-        logger.info("Processed upload documents from request")
-    except KeyError as e:
-        logger.error(e, {"Result": f"Upload Unsuccessful due to {str(e)}"})
         return ApiGatewayResponse(
-            400, f"An error occurred due to missing key: {str(e)}", "GET"
+            e.status_code, f"An error occurred due to: {str(e.message)}", "POST"
         ).create_api_gateway_response()
-    except ValidationError as e:
-        logger.error(e, {"Result": f"Upload Unsuccessful due to {str(e)}"})
-        return ApiGatewayResponse(
-            400, "Failed to parse document upload request data", "GET"
-        ).create_api_gateway_response()
-    except JSONDecodeError as e:
-        logger.error(e, {"Result": f"Upload Unsuccessful due to {str(e)}"})
-        return ApiGatewayResponse(
-            400, f"Invalid json in body: {str(e)}", "GET"
-        ).create_api_gateway_response()
-    except InvalidResourceIdException as e:
-        logger.error(e, {"Result": f"Upload Unsuccessful due to {str(e)}"})
-        return ApiGatewayResponse(
-            400, "Invalid NHS number", "GET"
-        ).create_api_gateway_response()
-
-    s3_service = S3Service()
 
-    arf_documents: list[NHSDocumentReference] = []
-    lg_documents: list[NHSDocumentReference] = []
-    url_responses = {}
-
-    for document in upload_request_documents:
-        document_type = SupportedDocumentTypes.get_from_field_name(document.docType)
-        if document_type is None:
-            return ApiGatewayResponse(
-                400, "An error occurred processing the required document type", "POST"
-            ).create_api_gateway_response()
+    return ApiGatewayResponse(
+        200, json.dumps(docs_services.url_responses), "POST"
+    ).create_api_gateway_response()
 
-        logger.info("Provided document is supported")
 
-        s3_object_key = create_reference_id()
+def processing_event_details(event):
+    try:
+        body = json.loads(event.get("body", ""))
 
-        document_reference: NHSDocumentReference
+        if not body or not isinstance(body, dict):
+            raise CreateDocumentRefException(400, "Missing event body")
+        nhs_number = body.get("subject", {}).get("identifier", {}).get("value", None)
 
-        if document_type == SupportedDocumentTypes.LG.value:
-            document_reference = NHSDocumentReference(
-                nhs_number=nhs_number,
-                s3_bucket_name=lg_s3_bucket_name,
-                reference_id=s3_object_key,
-                content_type=document.contentType,
-                file_name=document.fileName,
+        if not nhs_number:
+            raise CreateDocumentRefException(
+                400, "Request body missing some properties"
```

Comment:
Instead of calling multiple .get(), I think we could just try to drill down the nested object and catch all KeyError & TypeError. That would give better readability and is more aligned with the EAFP principle of python.

For example we could do it like:

```
def processing_event_details(event):
    try:
        body = json.loads(event["body"])

        if not body or not isinstance(body, dict):
            raise CreateDocumentRefException(400, "Missing request body")

        nhs_number = body["subject"]["identifier"]["value"]
        content = body["content"][0]
        doc_list = content["attachment"]

        return nhs_number, doc_list

    except (JSONDecodeError, AttributeError) as e:
        logger.error(e)
        raise CreateDocumentRefException(400, f"Invalid json in body: {str(e)}")
    except (KeyError, TypeError) as e:
        logger.error(e)
        raise CreateDocumentRefException(400, "Request body missing some properties")
```

Reply:
Done